testing/dynamodb/integration_test.go

```diff
@@ -0,0 +1,244 @@
+		AttributeDefinitions: []*dydb.AttributeDefinition{
+			{
+				AttributeName: aws.String("Year"),
+				AttributeType: aws.String("S"),
+			},
+			{
+				AttributeName: aws.String("Title"),
+				AttributeType: aws.String("S"),
+			},
+		},
+		KeySchema: []*dydb.KeySchemaElement{
+			{
+				AttributeName: aws.String("Year"),
+				KeyType:       aws.String("HASH"),
+			},
+			{
+				AttributeName: aws.String("Title"),
+				KeyType:       aws.String("RANGE"),
+			},
+		},
+		ProvisionedThroughput: &dydb.ProvisionedThroughput{
+			ReadCapacityUnits:  aws.Int64(10),
+			WriteCapacityUnits: aws.Int64(10),
+		},
+		TableName: aws.String(tableName),
+	}
+	_, err := dydbClient.CreateTable(createTableInput)
+	if err != nil {
+		t.Fatalf("Got error calling CreateTable: %s", err)
+	}
+
+	item := Item{
+		Year:   "2015",
+		Title:  "The Big New Movie",
+		Plot:   "Nothing happens at all.",
+		Rating: 0.0,
+	}
+	av, err := dynamodbattribute.MarshalMap(item)
+	if err != nil {
+		t.Fatalf("Got error marshalling new movie item: %s", err)
+	}
+
+	putItemInput := &dydb.PutItemInput{
+		Item:      av,
+		TableName: aws.String(tableName),
+	}
+
+	_, err = dydbClient.PutItem(putItemInput)
+	if err != nil {
+		t.Fatalf("Got error calling PutItem: %s", err)
+	}
+	log.Println("Successfully created table and put item for dynamodb")
+}
+
+func TestIntegration_DYNAMODB_SimpleUse(t *testing.T) {
+	onlyRunForEmulatorTest(t)
+	t.Parallel()
+
+	tmpdir := prepareIntegrationTest(t)
+	defer os.RemoveAll(tmpdir)
+
+	now := time.Now()
+	dbName, _ := conversion.GetDatabaseName(conversion.DYNAMODB, now)
+	dbPath := fmt.Sprintf("projects/%s/instances/%s/databases/%s", projectID, instanceID, dbName)
+	filePrefix := filepath.Join(tmpdir, dbName+".")
+
+	err := cmd.CommandLine(conversion.DYNAMODB, projectID, instanceID, dbName, false, false, false, 0, "", &conversion.IOStreams{Out: os.Stdout}, filePrefix, now)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Drop the database later.
+	defer dropDatabase(t, dbPath)
+
+	checkResults(t, dbPath)
+}
+
+func checkResults(t *testing.T, dbPath string) {
+	// Make a query to check results.
+	ctx := context.Background()
+	client, err := spanner.NewClient(ctx, dbPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	checkBigInt(ctx, t, client)
+}
+
+func checkBigInt(ctx context.Context, t *testing.T, client *spanner.Client) {
+	stmt := spanner.Statement{SQL: `SELECT Year, Title, Plot FROM Movies`}
+	iter := client.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	for {
+		row, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Println("Error reading row: ", err)
+			t.Fatal(err)
+			break
+		}
+		var year, title, plot string
+		if err := row.Columns(&year, &title, &plot); err != nil {
+			log.Println("Error reading into variables: ", err)
+			t.Fatal(err)
+			break
+		}
+		log.Printf("Found the following:\nYear: %s\nTitle: %s\nPlot: %s\n", year, title, plot)
```

Comment:
I am assuming this will get cleaned up in a follow up PR with actual comparison between Dynamodb and Spanner rows.

Reply:
Correct let me fix that in this only